node/sc/bridge_manager.go

```diff
@@ -431,21 +435,40 @@ func (bi *BridgeInfo) GetCurrentBlockNumber() (uint64, error) {
 
 // DecodeRLP decodes the Klaytn
 func (b *BridgeJournal) DecodeRLP(s *rlp.Stream) error {
-	var elem struct {
+	var LegacyBridgeAddrInfo struct {
 		LocalAddress  common.Address
 		RemoteAddress common.Address
 		Paired        bool
 	}
-	if err := s.Decode(&elem); err != nil {
-		return err
+	var BridgeAddrInfo struct {
+		BridgeAlias   string
+		LocalAddress  common.Address
+		RemoteAddress common.Address
+		Paired        bool
+	}
+	if !b.isLegacyBridgeJournal {
+		if err := s.Decode(&BridgeAddrInfo); err != nil {
```

Comment:
Don't we need to decode once again with legacy one?

Reply:
Do you consider a circumstance for lower compatibility so that a node with a lower version can read the aliased journal? If so, please check [here](https://github.com/klaytn/klaytn/pull/1277#discussion_r840261142) and add an opinion again here.

Reply:
Hmm.. Okay, but I think it is quite complicated. Why don't we try to decode once again here? instead of using isLegacyBridgeJournal? like this?
https://github.com/klaytn/klaytn/blob/dev/blockchain/types/tx_internal_data_serializer.go#L67-L77

Reply:
The code you give seems to decode type first and then data. Suppose we are going to have this way. In that case, we need to reconstruct the bridge journal struct by decomposing the data structure (`LocalAddress`, `RemoteAddress`, and `Paired`) with a type that indicates bridge or legacy. It seems not bad in my view. Please add how do you think.

Reply:
No, it's not. It just decode once again in the DecodeRLP() if the decoding fails.

Reply:
Let me ask you how to do that without reconstructing? I understood as the decode function reads and moves the internal pointer. Suppose that we first read something as a unit of struct and failed due to an incompatible legacy struct. At this point, if we try to read with another structure again, the next struct is decoded and the first one was already gone away. That's why I did reset and read again with another struct. The pattern that decodes the type of struct and decodes the actual interesting data again looks good to me, as I said above.

Reply:
I retook a look at the decode failure logic. The code you provided works well even if the decoding failure exists. The reason is that the target failed type is scalar type (`uint16` = `TxType`). In our case, however, the failed type is a struct, not a scalar. The [struct decoder](https://github.com/klaytn/klaytn/blob/dev/rlp/decode.go#L432) decodes one by one with an iteration of all the fields. The problem is that the struct decoder successfully decodes some fields but not some. For example, there are two structures `struct A { a,b int}, struct B { a,b,c int} `. Say a target encoded structure is A and we try to decode with struct B. The properties `a` and `b` are successfully decoded but, not c and get an error. Thus, the next read always fails since the read pointer does not point to the proper starting position of the expected encoded structure.

Reply:
@hyunsooda hmm but it seems like the first field of `BridgeAddrInfo` is a string while the first field of `LegacyBridgeAddrInfo` is a common.address (byte[20]). Then I think when it tries to decode the first field, it should fail. What do you think?

Reply:
The first failure output is correct with your concern. From all the following failure outputs, the error `expected input list for struct { ... }` would continuously pop up. The struct decoder calls the` List()` function that does actually read, meaning that the next read is not a failed type again. 

Let me show you two concrete examples. The first example is a primitive type passing to the `Decode()` function. The second one is when a different type of struct is passed.

1) Failed to decode struct value with a primitive type, but never do read anything actually. Next time we call `Decode()` again with a proper struct type and it succeeds. 

2) Failed to decode struct value with a primitive type and have read that size of the struct. Next time we call `Decode()` again with a proper struct type and fail due to the incorrect read pointer (i.e., malformed read pointer).   

If you have a doubt still, I recommend you to test and debug these two cases above.

Reply:
Got it. Understood. I think we can do it here like what you did [here](https://github.com/klaytn/klaytn/pull/1277/files#diff-5354626cdfa967dd6b4dbe6fd55e43f5bc198018bb25bb38fc6d21b18fab9f2eR95-R104).

Is there any differences?

Reply:
@hyunsooda Okay. I got it. Let's go then what it is.